src/algebra/cubic_discriminant.lean

```diff
@@ -0,0 +1,277 @@
+/-
+Copyright (c) 2022 David Kurniadi Angdinata. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata
+-/
+
+import field_theory.splitting_field
+
+/-!
+# Cubics and discriminants
+
+This file defines cubic polynomials over a semiring and their discriminants over a splitting field.
+
+## Main definitions
+
+* `cubic`
+* `disc`
+
+## Main statements
+
+* `disc_ne_zero_iff_roots_nodup`
+
+## References
+
+* <https://en.wikipedia.org/wiki/Cubic_equation>
+* <https://en.wikipedia.org/wiki/Discriminant>
+
+## Tags
+
+cubic, polynomial, root, discriminant
+-/
+
+noncomputable theory
+
+/-- The structure representing a cubic polynomial. -/
+structure cubic (R : Type*) [has_zero R] := (a b c d : R)
+
+instance (R : Type*) [has_zero R] : inhabited (cubic R) := ⟨⟨0, 0, 0, 0⟩⟩
+
+namespace cubic
+
+open cubic polynomial
+
+----------------------------------------------------------------------------------------------------
```

Comment:
We usually don't have such delimiters in mathlib. I think the section heading on the line below is sufficient.
```suggestion
```